src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -510,20 +556,25 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
 
         const isMyMembership = (m: CallMembership): boolean =>
             m.sender === this.client.getUserId() && m.deviceId === this.client.getDeviceId();
-        const callMembersChanged =
-            oldMemberships
-                .filter((m) => !isMyMembership(m))
-                .map(getParticipantIdFromMembership)
-                .sort()
-                .join() !==
-            this.memberships
-                .filter((m) => !isMyMembership(m))
-                .map(getParticipantIdFromMembership)
-                .sort()
-                .join();
-
-        if (callMembersChanged && this.isJoined()) {
-            this.requestKeyEventSend();
+
+        if (this.isJoined() && this.makeNewKeyTimeout === undefined) {
```

Comment:
If I got it correctly, in the current implementation everyone must leave within 3 seconds.  Should the timeout not better be extended here to wait for another amount of time? So that keys are send when there was no leave for 3 seconds?

Reply:
I didn't wan't to have a leave extend the period since I guess this could, in theory, be used to prevent a participant form ever changing their keys. This seemed simple & good as a start: it's not critical if keys have to be rotated again.